MessageBird/RequestValidator.cs

```diff
@@ -0,0 +1,154 @@
+        ///         <description>a unique token ID to implement an optional non-replay check (NOT validated by default).</description>
+        ///     </item>
+        ///     <item>
+        ///         <term>nbf</term>
+        ///         <description>the not before timestamp.</description>
+        ///     </item>
+        ///     <item>
+        ///         <term>exp</term>
+        ///         <description>the expiration timestamp is ensuring that a request isn't captured and used at a later time.</description>
+        ///     </item>
+        ///     <item>
+        ///         <term>iss</term>
+        ///         <description>the issuer name, always MessageBird.</description>
+        ///     </item>
+        /// </list>
+        /// </summary>
+        /// 
+        /// <param name="signature">request signature taken from request header MessageBird-Signature-JWT.</param>
+        /// <param name="url">full URL of the receiving endpoint, including scheme, host, path and query.</param>
+        /// <param name="body">request body.</param>
+        /// <returns>A dictionary representing decoded JWT signature token claims</returns>
+        /// <exception cref="RequestValidationException">Throws if signature can not be verified.</exception>
+        public IDictionary<string, string> ValidateSignature(string signature, string url, byte[] body)
+        {
+            IDictionary<string, string> payload;
+            try
+            {
+                payload = _jwtBuilder.Decode<IDictionary<string, string>>(signature);
+            }
+            catch (Exception e)
+            {
+                throw new RequestValidationException(e.Message, e);
+            }
+
+            string[] requiredClaims = { "iss", "nbf", "exp" };
+            foreach (string claim in requiredClaims)
+            {
+                if (payload.ContainsKey(claim)) continue;
+                throw new RequestValidationException(String.Format("invalid jwt: claim {0} is missing", claim));
+            }
+
+            if (!payload["iss"].Equals("MessageBird"))
+            {
+                throw new RequestValidationException("invalid jwt: claim iss has wrong value");
+            }
+
+            if (!_skipURLValidation)
+            {
+                string expectedURLHash = GetSHA256Hash(Encoding.UTF8.GetBytes(url));
+                if (!payload["url_hash"].Equals(expectedURLHash))
+                {
+                    throw new RequestValidationException("invalid jwt: claim url_hash is invalid");
+                }
+            }
+
+            bool bodyExist = body != null && body.Length > 0;
+            bool payloadHashExist = payload.ContainsKey("payload_hash");
+            if (!bodyExist && payloadHashExist)
+            {
+                throw new RequestValidationException("invalid jwt: claim payload_hash is set but actual payload is missing");
+            }
+            else if (bodyExist && !payloadHashExist)
```

Comment:
Since the previous exception wouild break the flow anyway, no need to chain those statements with `else`.